codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -679,29 +617,18 @@ final void takeChild(DefaultStream child, boolean exclusive, List<ParentChangedE
                 events.add(new ParentChangedEvent(child, oldParent));
                 notifyParentChanging(child, this);
                 child.parent = this;
-                // We need the removal operation to happen first so the prioritizableForTree for the old parent to root
-                // path is updated with the correct child.prioritizableForTree() value. Note that the removal operation
-                // may not be successful and may return null. This is because when an exclusive dependency is processed
-                // the children are removed in a previous recursive call but the child's parent link is updated here.
-                if (oldParent != null && oldParent.children.remove(child.id()) != null &&
-                        !child.isDescendantOf(oldParent)) {
-                    oldParent.decrementPrioritizableForTree(child.prioritizableForTree());
-                    if (oldParent.prioritizableForTree() == 0) {
-                        // There are a few risks with immediately removing nodes from the priority tree:
-                        // 1. We are removing nodes while we are potentially shifting the tree. There are no
-                        // concrete cases known but is risky because it could invalidate the data structure.
-                        // 2. We are notifying listeners of the removal while the tree is in flux. Currently the
-                        // codec listeners make no assumptions about priority tree structure when being notified.
-                        removeStream(oldParent, null);
-                    }
+                // Note that the removal operation may not be successful and may return null. This is because when an
+                // exclusive dependency is processed the children are removed in a previous recursive call but the
+                // child's parent link is updated here.
+                if (oldParent != null) {
+                    oldParent.children.remove(child.id());
```

Comment:
It's been a while since I looked closely at this code, so bare with me...

If this is being done by a tree restructure operation, can we get into trouble by removing immediately?